### ISSUE TITLE
Istio proxy container is failing to start with Failed to create directory for ./var/run/secrets/workload-spiffe-uds/socket: mkdir var/run/secrets/workload-spiffe-uds: read-only file system

### ISSUE THREAD

**Issue by malovme** [istio/istio#38217]
### Bug Description

```
2022-04-01T11:03:21.235838Z info  JWT policy is third-party-jwt
2022-04-01T11:03:21.235843Z info  using credential fetcher of JWT type in cluster.local trust domain
2022-04-01T11:03:22.256088Z info  SDS socket not found. Starting Istio SDS Server
2022-04-01T11:03:22.256131Z info  CA Endpoint istiod.istio-system.svc:15012, provider Citadel
2022-04-01T11:03:22.256104Z info  Opening status port 15020
2022-04-01T11:03:22.256156Z info  Using CA istiod.istio-system.svc:15012 cert with certs: var/run/secrets/istio/root-cert.pem
2022-04-01T11:03:22.256251Z info  citadelclient Citadel client using custom root cert: var/run/secrets/istio/root-cert.pem
2022-04-01T11:03:22.274596Z info  ads All caches have been synced up in 1.044429289s, marking server ready
2022-04-01T11:03:22.274803Z warn  Failed to create directory for ./var/run/secrets/workload-spiffe-uds/socket: mkdir var/run/secrets/workload-spiffe-uds: read-only file system
2022-04-01T11:03:22.274849Z error sds Failed to set up UDS path: failed to listen on unix socket "./var/run/secrets/workload-spiffe-uds/socket": listen unix ./var/run/secrets/workload-spiffe-uds/socket: bind: no such file or directory
2022-04-01T11:03:22.274855Z info  sds SDS server for workload certificates started, listening on "./var/run/secrets/workload-spiffe-uds/socket"
2022-04-01T11:03:22.274869Z info  xdsproxy  Initializing with upstream address "istiod.istio-system.svc:15012" and cluster "Kubernetes"
2022-04-01T11:03:22.276579Z info  sds Starting SDS grpc server
2022-04-01T11:03:22.276668Z warn  Failed to create directory for ./var/run/secrets/workload-spiffe-uds/socket: mkdir var/run/secrets/workload-spiffe-uds: read-only file system
2022-04-01T11:03:22.276724Z error sds SDS grpc server for workload proxies failed to set up UDS: failed to listen on unix socket "./var/run/secrets/workload-spiffe-uds/socket": listen unix ./var/run/secrets/workload-spiffe-uds/socket: bind: no such file or directory
2022-04-01T11:03:22.277375Z info  starting Http service at 127.0.0.1:15004
2022-04-01T11:03:22.278571Z info  Pilot SAN: [istiod.istio-system.svc]
2022-04-01T11:03:22.280450Z info  Starting proxy agent
2022-04-01T11:03:22.280480Z info  Epoch 0 starting
2022-04-01T11:03:22.280500Z info  Envoy command: [-c etc/istio/proxy/envoy-rev0.json --restart-epoch 0 --drain-time-s 45 --drain-strategy immediate --parent-shutdown-time-s 60 --local-address-ip-version v4 --file-flush-interval-msec 1000 --disable-hot-restart --log-format %Y-%m-%dT%T.%fZ  %l  envoy %n  %v -l warning --component-log-level misc:error]
2022-04-01T11:03:23.277451Z warn  Failed to create directory for ./var/run/secrets/workload-spiffe-uds/socket: mkdir var/run/secrets/workload-spiffe-uds: read-only file system
2022-04-01T11:03:23.277582Z error sds SDS grpc server for workload proxies failed to set up UDS: failed to listen on unix socket "./var/run/secrets/workload-spiffe-uds/socket": listen unix ./var/run/secrets/workload-spiffe-uds/socket: bind: no such file or directory
2022-04-01T11:03:23.283586Z warn  ca  ca request failed, starting attempt 1 in 103.583184ms
2022-04-01T11:03:23.387848Z warn  ca  ca request failed, starting attempt 2 in 205.386114ms
2022-04-01T11:03:23.593371Z warn  ca  ca request failed, starting attempt 3 in 429.688774ms
2022-04-01T11:03:24.024024Z warn  ca  ca request failed, starting attempt 4 in 723.617885ms
2022-04-01T11:03:25.278832Z warn  Failed to create directory for ./var/run/secrets/workload-spiffe-uds/socket: mkdir var/run/secrets/workload-spiffe-uds: read-only file system
2022-04-01T11:03:25.278994Z error sds SDS grpc server for workload proxies failed to set up UDS: failed to listen on unix socket "./var/run/secrets/workload-spiffe-uds/socket": listen unix ./var/run/secrets/workload-spiffe-uds/socket: bind: no such file or directory
2022-04-01T11:03:27.808166Z info  cache generated new workload certificate  latency=5.533264895s ttl=23h59m59.191843153s
2022-04-01T11:03:27.808203Z info  cache Root cert has changed, start rotating root cert
2022-04-01T11:03:27.808219Z info  ads XDS: Incremental Pushing:0 ConnectedEndpoints:0 Version:
2022-04-01T11:03:27.808282Z info  cache returned workload trust anchor from cache ttl=23h59m59.191720438s
2022-04-01T11:03:29.279726Z warn  Failed to create directory for ./var/run/secrets/workload-spiffe-uds/socket: mkdir var/run/secrets/workload-spiffe-uds: read-only file system
2022-04-01T11:03:29.279814Z error sds SDS grpc server for workload proxies failed to set up UDS: failed to listen on unix socket "./var/run/secrets/workload-spiffe-uds/socket": listen unix ./var/run/secrets/workload-spiffe-uds/socket: bind: no such file or directory
2022-04-01T11:03:29.662011Z info  xdsproxy  connected to upstream XDS server: istiod.istio-system.svc:15012
2022-04-01T11:03:37.280480Z warn  Failed to create directory for ./var/run/secrets/workload-spiffe-uds/socket: mkdir var/run/secrets/workload-spiffe-uds: read-only file system
2022-04-01T11:03:37.280569Z error sds SDS grpc server for workload proxies failed to set up UDS: failed to listen on unix socket "./var/run/secrets/workload-spiffe-uds/socket": listen unix ./var/run/secrets/workload-spiffe-uds/socket: bind: no such file or directory
2022-04-01T11:03:53.281441Z warn  sds SDS grpc server could not be started
2022-04-01T11:04:01.334132Z warning envoy config  StreamSecrets gRPC config stream closed since 31s ago: 14, upstream connect error or disconnect/reset before headers. reset reason: connection failure, transport failure reason: immediate connect error: No such file or directory
```

### Version

```prose
Image:         gcr.io/istio-testing/proxyv2:latest
Image ID:      docker-pullable://gcr.io/istio-testing/proxyv2@sha256:c1a575c54eef3b8642ceca205c4f2551eda026d7a910e5b2330d8ab6613d366a
```

**Comment by malovme** [comment]
Might be related to daa835c3f63218e57a6ec701d477f732e6dfaa47

**Comment by howardjohn** [comment]
Sounds like you are running `gcr.io/istio-testing/proxyv2:latest` but didn't pull in the new helm charts. The two are coupled

**Comment by timscriv** [comment]
We were using the Helm Charts for istiod pointed to the tag 1.13.2. When the istio-testing was pushed to the container registry, inside the 1.13.2 helm chart there was a reference to `latest` which caused all new pods to not be able to connect since istiod was erroring with the above error. This required us to temporarily point to the master charts since they are the only ones with the updates to support the latest changes. 

This error occurred due to a reference in the helm charts to latest that we have no control over. Can this be pointed to a stable tag instead of latest so it isn't changed out from under us?

**Comment by FRABUCHI** [comment]
We are experiencing the same phenomenon.
<img width="1890" alt="스크린샷 2022-04-04 오후 12 33 48" src="https://user-images.githubusercontent.com/37655730/161469544-d3860a5a-1c6b-44c8-85fb-f1ea4e1835e9.png">

I lowered the version to 1.20, but the same phenomenon still appears.

**Comment by FRABUCHI** [comment]
> 우리는 태그 1.13.2를 가리키는 istiod에 대해 Helm 차트를 사용하고 있었습니다. istio-testing이 컨테이너 레지스트리로 푸시되었을 때 1.13.2 helm 차트 내부에 `latest`istiod가 위의 오류로 오류가 발생했기 때문에 모든 새 포드를 연결할 수 없도록 하는 참조가 있었습니다. 최신 변경 사항을 지원하는 업데이트가 있는 유일한 차트이기 때문에 마스터 차트를 일시적으로 가리켜야 했습니다.
> 
> 이 오류는 우리가 제어할 수 없는 최신에 대한 helm 차트의 참조로 인해 발생했습니다. 이것이 우리 아래에서 변경되지 않도록 최신 대신 안정적인 태그를 가리킬 수 있습니까?

what version is stable?

**Comment by haksuuu** [comment]
I also faced this problem. Anyone help!

**Comment by beaudeanadams** [comment]
Anyone facing this issue, add this to your helm values, and you will use the dockerhub istio repo instead. 
Ensure you set 'tag' under global to be the same version as the helm chart (i.e. 1.13.2)

```
global:
  hub: docker.io/istio
  tag: 1.13.2
```

**Comment by haksuuu** [comment]
@beaudeanadams What I have to put that global value?
in my application helm chart value?

or istiod application? 

thank you

**Comment by FRABUCHI** [comment]
> Anyone facing this issue, add this to your helm values, and you will use the dockerhub istio repo instead. Ensure you set 'tag' under global to be the same version as the helm chart (i.e. 1.13.2)
> 
> ```
> global:
>   hub: docker.io/istio
>   tag: 1.13.2
> ```

Good solution. I soleved issue by adding tag on istiod application hat you suggest.

**Comment by timscriv** [comment]
> > 우리는 태그 1.13.2를 가리키는 istiod에 대해 Helm 차트를 사용하고 있었습니다. istio-testing이 컨테이너 레지스트리로 푸시되었을 때 1.13.2 helm 차트 내부에 `latest`istiod가 위의 오류로 오류가 발생했기 때문에 모든 새 포드를 연결할 수 없도록 하는 참조가 있었습니다. 최신 변경 사항을 지원하는 업데이트가 있는 유일한 차트이기 때문에 마스터 차트를 일시적으로 가리켜야 했습니다.
> 
> > 
> 
> > 이 오류는 우리가 제어할 수 없는 최신에 대한 helm 차트의 참조로 인해 발생했습니다. 이것이 우리 아래에서 변경되지 않도록 최신 대신 안정적인 태그를 가리킬 수 있습니까?
> 
> 
> 
> what version is stable?

I was referring to a stable tag on the container (e.g. 1.13.2) instead of using latest by default. 

Currently we are testing the fix from the above messages to see if this resolves.

**Comment by visteras** [comment]
I checked my istio-ingress/egress/base/discovery values, and i see in everyone:
```
global:
  hub: docker.io/istio
  tag: 1.13.2
```
where i should set this value? link or chart name please?

**Comment by timscriv** [comment]
To get rid of this completely we had to do these values (they are sub-helm charts) but it should be clear enough.

```
istioCni:
  targetRevision: 1.13.2
  values:
    cni:
      hub: docker.io/istio
      tag: 1.13.2
```
---
```
istiod:
  targetRevision: 1.13.2
  values:
    global:
      hub: docker.io/istio
      tag: 1.13.2
      proxy_init:
        image: proxyv2:1.13.2
```
---
```
istioEgress:
  targetRevision: 1.13.2
  values:
    global:
      hub: docker.io/istio
      tag: 1.13.2
```
---
```
istioIngress:
  targetRevision: 1.13.2
  values:
    global:
      hub: docker.io/istio
      tag: 1.13.2
```
---
> Removed settings that aren't related.

**Comment by FRABUCHI** [comment]
> I checked my istio-ingress/egress/base/discovery values, and i see in everyone:
> 
> ```
> global:
>   hub: docker.io/istio
>   tag: 1.13.2
> ```
> 
> where i should set this value? link or chart name please?

https://github.com/istio/istio/blob/master/manifests/charts/istio-control/istio-discovery/values.yaml

**Comment by ericvn** [comment]
I'm wondering if this is a problem because the manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml still has a `tag: latest` in it. Normally, the tag would have been updated to `1.13.2` by the build tooling. I did see this error in the istio.io testing for the helm charts because the test was incorrectly using `latest`. I've now updated the tests to use use the actual commit based on the istio commit used for the helm testing.

**Comment by ericvn** [comment]
I have found a failure in the release-building code to update the `gem-istio.yaml` file replacing the hub and tag fails (that file uses quotes which the other files do not).

The work around for now would be to add `--set global.tag=1.13.2 --set global.hub=docker.io/istio` to the helm installs.

I'm going to work on a code change for the next release so the files are updated correctly.

**Comment by ericvn** [comment]
I haven't been able to recreate the failure using the actual releases, just when using the source code directly. Would someone be able to articulate how you are installing Istio to see this failure. If I can recreate, I can verify a fix.

Thanks.

**Comment by bryankaraffa** [comment]
@ericvn  -- hopefully this is helpful.. pulled values and helm releases where I can reproduce this.. basically `v1.13.2` helm charts with `global.tag=latest` and `global.hub=gcr.io/istio-testing`

<details>

```sh
# Istio System Namespace [istio-base, istiod]
# Chart v1.13.2 from istio helm chart repo
$ helm ls -n istio-system
NAME                                            NAMESPACE       REVISION        UPDATED                                 STATUS          CHART                           APP VERSION
istio-base                                      istio-system    1               2022-04-04 20:14:57.5453432 +0000 UTC   deployed        base-1.13.2                     1.13.2     
istiod                                          istio-system    1               2022-04-04 20:15:12.307586305 +0000 UTC deployed        istiod-1.13.2                   1.13.2     

# Istio Ingress Namespace [gateway charts]
# Chart v1.13.2 from istio helm chart repo
$ helm ls -n istio-ingress
NAME                                    NAMESPACE       REVISION        UPDATED                                 STATUS          CHART                           APP VERSION
istio-privategateway                    istio-ingress   1               2022-04-04 20:15:58.011610327 +0000 UTC deployed        gateway-1.13.2                  1.13.2     
istio-publicgateway                     istio-ingress   1               2022-04-04 20:15:56.866830851 +0000 UTC deployed        gateway-1.13.2                  1.13.2             

# Istio-base has no user-supplied values
$ helm -n istio-system get values istio-base
USER-SUPPLIED VALUES:
null

# Istiod overrides 
# global.hub = gcr.io/istio-testing
# global.tag = latest
$ helm -n istio-system get values istiod
USER-SUPPLIED VALUES:
global:
  hub: gcr.io/istio-testing
  proxy:
    resources:
      requests:
        cpu: 10m
        memory: 64Mi
  tag: latest
meshConfig:
  accessLogFile: /dev/stdout
pilot:
  resources:
    requests:
      cpu: 50m
      memory: 200Mi
sidecarInjectorWebhook:
  neverInjectSelector:
  - matchExpressions:
    - key: app
      operator: In
      values:
      - kube-prometheus-stack-admission-create
      - kube-prometheus-stack-admission-patch

# Ingress Gateway overrides 
# global.hub = gcr.io/istio-testing
# global.tag = latest
$ helm -n istio-ingress get values istio-privategateway
USER-SUPPLIED VALUES:
global:
  hub: gcr.io/istio-testing
  tag: latest
labels:
  topology.istio.io/network: vpc-0cfa7da64035c0ef7
service:
  annotations:
    service.beta.kubernetes.io/aws-load-balancer-cross-zone-load-balancing-enabled: "true"
    service.beta.kubernetes.io/aws-load-balancer-internal: "true"
    service.beta.kubernetes.io/aws-load-balancer-type: nlb
```

</detail>

**Comment by bryankaraffa** [comment]
I dug into how the `global.tag` and `global.hub` values are used in the other helm charts, and it appears like `gateway` chart uses [different strategy for setting the image on deployment spec](https://github.com/istio/istio/blob/master/manifests/charts/gateway/templates/deployment.yaml#L46-L47) [and does not use `global{}` in [values.yaml](https://github.com/istio/istio/blob/master/manifests/charts/gateway/values.yaml) or have a way to declare that via values]

compared to the `istio-discovery` [[deployment](https://github.com/istio/istio/blob/master/manifests/charts/istio-control/istio-discovery/templates/deployment.yaml#L80-L84), [values](https://github.com/istio/istio/blob/master/manifests/charts/istio-control/istio-discovery/values.yaml#L241-L246)] and legacy `gateways` [[deployment](https://github.com/istio/istio/blob/master/manifests/charts/gateways/istio-ingress/templates/deployment.yaml#L92-L96), [values](https://github.com/istio/istio/blob/master/manifests/charts/gateways/istio-ingress/values.yaml#L156-L162)]  charts

**Comment by ericvn** [comment]
@bryankaraffa I understand how to see the error by forcing the `latest` image and test repo. I'm wondering how people are encountering this problem without forcing the incorrect image.

I suspect there is something in one of the manifest files that is causing the issue, possibly https://github.com/istio/istio/blob/release-1.13/manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml#L165 and similarly for the hub is a line further up. These lines aren't updated to the normal release values in the pushed helm charts as far as I can tell. I'm not sure if updating those values will resolve the issue or not.

**Comment by ericvn** [comment]
> We were using the Helm Charts for istiod pointed to the tag 1.13.2. When the istio-testing was pushed to the container registry, inside the 1.13.2 helm chart there was a reference to `latest` which caused all new pods to not be able to connect since istiod was erroring with the above error. This required us to temporarily point to the master charts since they are the only ones with the updates to support the latest changes.
> 
> This error occurred due to a reference in the helm charts to latest that we have no control over. Can this be pointed to a stable tag instead of latest so it isn't changed out from under us?

@timscriv Can you point to where the `latest` is in your 1.13.2 helm charts? I'm curious if it's the value you mentioned above.

**Comment by FRABUCHI** [comment]
> > I checked my istio-ingress/egress/base/discovery values, and i see in everyone:
> > ```
> > global:
> >   hub: docker.io/istio
> >   tag: 1.13.2
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > where i should set this value? link or chart name please?
> 
> https://github.com/istio/istio/blob/master/manifests/charts/istio-control/istio-discovery/values.yaml
> 
> Also, I think you should change this value as @timscriv wrote it would be good. When tag of proxyv2 image is latest, the problem is happend.
> 
> ```
> istiod:
>   targetRevision: 1.13.2
>   values:
>     global:
>       hub: docker.io/istio
>       tag: 1.13.2
>       proxy_init:
>         image: proxyv2:1.13.2
> ```

He maybe found in https://github.com/istio/istio/blob/master/manifests/charts/istio-control/istio-discovery/values.yaml

**Comment by ericvn** [comment]
@FRABUCHI You are pointing at the source branch and there the value is `latest` which makes sense. When a release is built, the `latest` values are changed in the manifest files for that release. If you pull the 1.13.2 release and look in the manifest files, you see the above values. This also makes sense. I did find one instance where a manifest file has `latest` in the release manifest, but it sounds like that doesn't have a negative affect on the release.

So you only see the problem when using `latest` with an older set of helm charts which are incompatible. This is also correct. If you want to use `latest` you need to be using a compatible set of helm charts, which you can get from the master branch, and that should work.

**Comment by timscriv** [comment]
I think this is a correct statement but not intuitive if you are pointing to a tag and using a GitOps tool like Argo for helm. I think the issue of that we are pointing to the manifests that are in the Git repo and not the published charts. This might be expected but maybe there is a way to make it more clear that if you are using GitOps tooling that it won't work as expected since there are some tasks that change the helm charts before they are published to the chart repo and that it is required to use that. I am double checking all that today to see if it clears all the issues.

**Comment by ericvn** [comment]
If you are using version 1.13 charts with 1.14 images, I would expect that there would be occasional hiccups like this.

**Comment by howardjohn** [comment]
Catching up so sorry if this is repeated - the only supported helm charts are the official helm repository (highly recommended, only one documented on istio.io) or the release tar.gz.

Pulling from git, or any other method that uses `latest` is not supported

*if you hit this issue you are installing in a dangerous an unsupported way*

**Comment by albertschwarzkopf** [comment]
Hi @malovme. I have noticed the same behaviour yesterday. I have upgraded Istio from 1.11.4 to the latest version (1.14.1) and the istio-proxies had the same error. 

Istio 1.13.5 is working as expected. I think this is related to this change here: https://github.com/istio/istio/commit/daa835c3f63218e57a6ec701d477f732e6dfaa47 and the fact that my worker-nodes are running on Bottlerocket OS (with read-only root filesystem)?

Kubernetes Version: 1.21 AWS EKS, Nodes are running on Bottlerocket OS (with read-only root filesystem)

**Comment by spstarr** [comment]
This should be reopened against 

```
$ curl -L https://istio.io/downloadIstio | sh -
$ cd istio-1.14.2
```

**Comment by hzxuzhonghu** [comment]
@spstarr Did you build your own docker image?

**Comment by zander312** [comment]
@hzxuzhonghu @spstarr running into this issue as well. We are trying to upgrade from 1.13.5. We are using tetrates distroless images. `1.14.1-tetratefips-v0-distroless` https://istio.tetratelabs.io/

pulled down the Istio 1.14 manifests and used them for the install so manifests should be compatible.

any tips on how we can further debug this?

**Comment by lam-man** [comment]
Any progress on this issue? We also saw this issue while upgrading Istio from 1.13.4 to 1.14.2. For the `gen-istio.yaml`, we are not using `1.14.2` instead of the `latest` tag.

**Comment by lam-man** [comment]
Hi Community. Is there anyone trying to find out the cause of this issue? I would more than glad to have a discussion with you. Currently, we verified on 3 version, respectively 1.14.1, 1.14.2, and 1.14.3 regarding the upgrade from 1.13.4. All version failed with this issue.

**Comment by howardjohn** [comment]
How did you install? THe only way this should be possible is by incorrectly installing leading to a mismatch of image and injection template.

**Comment by albertschwarzkopf** [comment]
In my case I'm using FluxCD for installing and upgrading applications. So in that case I switched the version from 1.11.4 to 1.14.1 for your chart manifests (https://github.com/istio/istio/tree/master/manifests/charts).

I have changed the image value in that case to:

`global:  
hub: gcr.io/istio-release  ``
    tag: 1.14.1`

**Comment by howardjohn** [comment]
So you are using the git repo directly? That is not supported. Use the official release tar.gz or the helm repository. The git state is for development.

**Comment by lam-man** [comment]
> How did you install? THe only way this should be possible is by incorrectly installing leading to a mismatch of image and injection template.

We are using the official tar.gz file for all three versions. I saw there is a fix posted. Will test this again. Thanks

**Comment by howardjohn** [comment]
If you are seeing this you have a mismatch between docker image versions and configmaps

**Comment by qudongfang** [comment]
I think this issue should be reopened.

I confirm this is also an issue on the VM with `1.4.3`.

`/var/run/secrets/workload-spiffe-uds` is missing since `/var/run` is wiped after VM reboot.

The `istio` process started fine if I manually run `mkdir -p /var/run/secrets`.

IMHO, We should at least support customizing the directory.

**Comment by howardjohn** [comment]
It's intended to be missing, it will create it.

**Comment by qudongfang** [comment]
Sorry, If I didn't describe the issue clearly.

I got the following error:
```
{"level":"warn","time":"2022-09-07T15:04:10.125120Z","msg":"Failed to create directory for ./var/run/secrets/workload-spiffe-uds/socket: mkdir var/run/secrets: permission denied"}`
```

Does that mean we need to give the `istio-proxy` user permission to `mkdir /var/run/secrets` somehow?

**Comment by howardjohn** [comment]
yes

**Comment by lam-man** [comment]
> If you are seeing this you have a mismatch between docker image versions and configmaps

Hi @howardjohn . We really appreciate your help here. I want to confirm that you are CORRECT on this. We found out the root cause of the issue with your response. Here are some details for people who has interest: 

- We install Istio using `istio-operator`. (This is not encouraged any more according to the doc. Details in [doc](https://istio.io/latest/docs/setup/install/operator/). 
- When installing Istio with `istio-operator`, we first install `istio-operator` and then `istio-operator` will help to install resources like `istiod` automatically. 
- We customized the `istio-operator` installing by adding helm hooks. The purpose of the hook is to restart resources with sidecar proxy injection.
- The problem comes when we run the restart helm hook. **When we run the restart helm hook, `istio-operator` hasn't finished the installation of `istiod` and `istio-gateway`, which cause the MISMATCH between docker image version and configmaps.** 

I hope this can help others with the same problem.

**Comment by howardjohn** [comment]
Discussed this on slack, root cause was using a sidecar injector configmap
from 1.10 (for some calico integration)

**Comment by akitok** [comment]
I had the same problem with this issue.
So I tried upgrading to the recently released [Istio 1.14.5](https://istio.io/latest/news/releases/1.14.x/announcing-1.14.5/) and the problem in this issue was resolved.
I hope this will be helpful to someone.

**Comment by morariu** [comment]
Hi. Is this fixed in 1.15 at all? I am not using helm, pulled the Istio images 1.14.5 from docker hub and installing with kubectl offline. How can I apply the fix in this case? Thanks

**Comment by zirain** [comment]
> installing with kubectl offline

what does this mean?

**Comment by zirain** [comment]
you install istio without istioctl and helm? how did you generate yaml?

**Comment by morariu** [comment]
> you install istio without istioctl and helm? how did you generate yaml?

I have generated the yaml files, that's not the problem. It worked for 1.11 version but facing issues with 1.14.5

Looking to use helm in this case, I have to re-package all the charts and push individually to ECR in AWS as only one chart per repo is supported. I found istio-base but cannot find istiod chart in the 1.14.5 release. The documentation mentions: **Install the Istio discovery chart which deploys the istiod service** however I can only see istiod-remote in /istio-1.14.5/manifests/charts/. List below.

base
default
gateway
gateways
index.yaml
install-OpenShift.md
istio-cni
istio-control
istiod-remote
istio-operator

 Am I missing anything?
